Foundation/NSThread.swift

```diff
@@ -57,18 +57,30 @@ private func NSThreadStart(_ context: UnsafeMutableRawPointer?) -> UnsafeMutable
 }
 
 open class Thread : NSObject {
-    
+    static internal let _mainThread = Thread(thread: pthread_self())
```

Comment:
@phausler 👍  following your suggestion i found that CF already has pthread_main_np shim implemented [here](https://github.com/apple/swift-corelibs-foundation/blob/173e9ea16cf1a0ed9b21cc11ea8a2d15b3f8dca9/CoreFoundation/Base.subproj/CFRuntime.c#L982)